### PR TITLE
CI: fixing doctests

### PR DESCRIPTION
**Pull request by andyfaff**
Travis is currently failing because doctests have some formatting issues due to [numpy 1.14](https://github.com/numpy/numpy/blob/master/doc/release/1.14.0-notes.rst#many-changes-to-array-printing-disableable-with-the-new-legacy-printing-mode).
From a ML conversation minor errors should be being picked up by the refguide checker, but it's not currently doing that.
This PR simply corrects the doctest output.

### CODE REVIEW

**Comment by pv** [comment]
Not related to this PR: that numpy prints complex numbers as 
`9.90012467+0.        j` looks annoying because that's not copypasteable 
any more. This maybe is a numpy bug...

**Comment by andyfaff** [comment]
> Not related to this PR: that numpy prints complex numbers as
`9.90012467+0.        j` looks annoying because that's not copypasteable
any more. This maybe is a numpy bug...

I agree, it doesn't look nice. The link in my first comment suggests that one can use a legacy print mode. Perhaps that's the way forward for the time being? I'm not sure how one would go about setting that mode, presumably in ref_guide.py?

**Comment by pv** [comment]
I think the documention has to agree with whatever Numpy does out of the 
box, so we shouldn't try to adjust the printing. The only thing could be 
to complain on numpy issue tracker that the new printing of complex 
numbers does not look optimal.

**Comment by andyfaff** [comment]
There's something going on with that specific example in tutorial/linalg.rst. The output is different on the travisCI box vs my OSX machine.

**Comment by andyfaff** [comment]
There's also an unrelated stats test failing on Appveyor, `test_weightedtau`.

**Comment by andyfaff** [comment]
see numpy/numpy#10355. @pv, I think you posted on there that we should hold off on this merge until the printing gets fixed. Unfortunately that'll mean that our CI is out of action until the next numpy release.

**Comment by xoviat** [comment]
The ultimate cause is that `np.min` throws a warning if a real number and `np.nan` are compared. Conceptually, this seems fine.

**Comment by andyfaff** [comment]
W.r.t `stats.weightedtau` I wonder why is it only showing up on a subset of the Windows tests?

In `stats.weightedtau`:
> NaNs are considered the smallest possible score.

The documentation for `np.amin` says:
> NaN values are propagated, that is if at least one item is NaN, the corresponding min value will be NaN as well.

It doesn't say that warnings are raised if NaN's are in the array. The code in [L3711 to L3714](https://github.com/scipy/scipy/blob/master/scipy/stats/stats.py#L3711) contains the lines:

```
    if np.isnan(np.min(y)):
            y = _toint64(y)
```

perhaps this should be:

```
if np.any(np.isnan(y)):
        y = _toint64(y)
```

**Comment by xoviat** [comment]
What happens with this on linux?

```
np.min([1, np.nan])
```

**Comment by andyfaff** [comment]
@xoviat it appears I edited your comment by mistake, sorry.

**Comment by xoviat** [comment]
The behavior appears to be deterministic on my windows machine; it always gives the warning (the results are the same). Do you think that, conceptually, there should be a warning?

**Comment by xoviat** [comment]
If so, then the test would appear to be broken.

**Comment by andyfaff** [comment]
I think the documentation and actual behaviour should tally.

> NaN values are propagated, that is if at least one item is NaN, the corresponding min value will be NaN as well.

This doesn't indicate that a warning will result. I have just done the following in a clean conda env (3.6.2) with numpy installed via pip. There were no warnings!

```
>>> import numpy as np
>>> np.version.version
'1.14.0'
>>> np.min([1., 2., 3., 4., np.nan])
nan
>>> np.min([1., 2., 3., np.nan, 4.])
nan
>>> np.min([1., 2., np.nan, 3., 4.])
nan
>>> np.min([1., np.nan, 2., 3., 4.])
nan
>>> np.min([np.nan, 1., 2., 3., 4.])
nan
>>> np.min([np.nan, 1.])
nan
>>> np.min([np.nan, 1., np.nan])
nan
>>> np.min([1., np.nan])
nan
>>> np.seterr(all='raise')
{'divide': 'warn', 'over': 'warn', 'under': 'ignore', 'invalid': 'warn'}
>>> np.min([1., np.nan])
nan
>>> np.min([np.nan, 1.])
nan
>>> np.min([np.nan, 1., 2., 3., 4.])
nan
>>> np.min([np.nan, 1., 2., 3., 4.])
nan
```

The weird behaviour in my previous post was that 

```
>>> np.min([1,2,np.nan])
nan
>>> with np.errstate(all='raise'):
...     np.min([1,2, np.nan])
... 
nan
```

didn't give any warnings, but then when I subsequently typed ```np.min([1., np.nan])```, there was a warning. Why should the length of the array change behaviour?

Something isn't working reproducibly across numpy installs.

**Comment by ev-br** [comment]
LGTM modulo a need for a rebate.

Refguide-checker should be able to handle some of these, but that needs someone to have a look. I probably will at some point, no ETA though.

File: doc/source/tutorial/fftpack.rst [hunk 1]
```diff
@@ -53,7 +53,7 @@ respectively as shown in the following example.
 >>> x = np.array([1.0, 2.0, 1.0, -1.0, 1.5])
 >>> y = fft(x)
 >>> y
-array([ 4.50000000+0.j        ,  2.08155948-1.65109876j,
+array([ 4.5       +0.        j,  2.08155948-1.65109876j,
```
Comment (pv) [inline doc/source/tutorial/fftpack.rst]: The complex printing format will apparently change again in numpy 1.14.1, so we may want to postpone proceeding once that's been fixed...

**Comment by pv** [comment]
Numpy 1.14.1 is out, so I think we could resume this discussion.

**Comment by andyfaff** [comment]
Rebased the PR, so we'll see what happens. AFAICT the doctesting is being done on py35, with pypi numpy?

Whilst looking at the travis config am I right in understanding that we test the numpy master branch in a py27 environment? We should probably move that to 3.x.

**Comment by andyfaff** [comment]
We obviously wouldn't have picked the numpy issues up with the doctesting unless we were targeting the master branch, rather than the released wheel.

**Comment by andyfaff** [comment]
A few minor nits needed changing.
What's the received wisdom on the doctest - we're obviously looking for correct numerical + object behaviour, but these tests are seem to be picky about the number of spaces, etc.

**Comment by ev-br** [comment]
While the default doctest behavior is very picky about whitespace, our test checker tries to select numeric entities and `np.assert_allclose` them, where it can. It just fails to find arrays nested too deep into other objects, and falls back to vanilla doctest. Which is arguably a deficiency in our doctester. 

I'd suggest to both fix the nits so that our examples match what numpy outputs out of the box, *and* improve the doctester (in a separate PR).

**Comment by andyfaff** [comment]
So if the changes I've made here finally work, then this should be mergeable. Currently PR's will have at least one fail because of the doctest errors fixed in this PR.

**Comment by person142** [comment]
Thanks @andyfaff; will be nice to have the CIs green again.

**Pull request merged**